### ISSUE TITLE
Disabling custom font sizes does not remove 'custom' option from Paragraph block font size dropdown

### ISSUE THREAD

**Issue by badlydrawnben** [WordPress/gutenberg#17941]
The dropdown of theme-defined font sizes also has a 'Custom' option, even though I have defined the theme font sizes and removed support for custom font sizes.

**To reproduce:**
Blank installation of WP 5.2.3 and TwentyNineteen theme.
Install and enable Gutenberg plugin (v6.6.0)
Add this snippet to functions.php
`add_theme_support('disable-custom-font-sizes');`

**Expected behaviour:**
The custom text size input field should not be shown - works as expected
The dropdown of text sizes (eg small, medium, large etc) should not also include 'Custom' as an option because that functionality has been removed by that snippet - Does not work as expected.

![image](https://user-images.githubusercontent.com/4472981/66772783-6c00e500-eeb5-11e9-9542-d9d03e15a638.png)

Note - this problem does not exist if the Gutenberg plugin is deactivated.

**Comment by Waqas-Ali-Azhar** [comment]
Hi @badlydrawnben , i reproduced the issue and tried adding a pull request here to resolve the issue.
https://github.com/WordPress/gutenberg/pull/18049

**Comment by gziolo** [comment]
Copying over from #18212 which turned out to be a duplicate:

> **Describe the bug**
> Originally reported in #18149:
> 
> > The example which has the custom sizes disabled works only partially, since, in the dropdown, there is still 
> >  an option labeled `Custom`:
> > ![Screen Shot 2019-10-31 at 10 40 33](https://user-images.githubusercontent.com/699132/67935891-0ad25480-fbcb-11e9-8bee-8f0ebc51172f.png)
> > 
> 
> **To reproduce**
> Steps to reproduce the behavior:
> 1. Run Storybook: `npm run design-system:dev` or open https://wordpress.github.io/gutenberg/design-system/components/.
> 2. Go to the `FontSizePicker` section.
> 3. Pick the one named `Without Custom Sizes`
> 4. Check the select box.
> 
> **Expected behavior**
> There is no `Custom` option when `disableCustomFontSizes` flag is passed to `FontSizePicker` component.
> 
> **Desktop (please complete the following information):**
> 
> - OS: macOS
> - Browser Chrome
> - Version 77

**Comment by mtias** [comment]
@gziolo is this still the case?

**Comment by gziolo** [comment]
We should have a different custom font size picker now, so it might be no longer a case. @sarayourfriend or @ciampo can you confirm?

At least in Storybook, it works as expected:

<img width="272" alt="Screen Shot 2021-07-19 at 13 51 54" src="https://user-images.githubusercontent.com/699132/126155882-0b2849a9-f298-4ca3-89ff-7d58fcb6b00c.png">

**Comment by ciampo** [comment]
> We should have a different custom font size picker now, so it might be no longer a case. @sarayourfriend or @ciampo can you confirm?

I'm personally not familiar with this component, but looking at the [folder history](https://github.com/WordPress/gutenberg/commits/trunk/packages/components/src/font-size-picker), there's been quite a few changes to its source code (refactors and bug fixes)

> At least in Storybook, it works as expected

I can confirm that the "Without Custom Sizes" storybook story seems to work as expected (and therefore doesn't reproduce the original bug in this issue)

**Comment by smileBeda** [comment]
This is frustrating.

It is by far the worst result off what bad naming conventions can do.

`add_theme_support( 'disable-custom-colors' );`
...?!

==> `remove_theme_support( 'custom-line-height' );` everyone gets it!
==> `add_theme_support( 'disable-custom-colors' );` ... 🫨 couldn't it just be named according to other such supports, namely, that would be `remove_theme_support( 'custom-colors' );`. Duh. Simple, and doc would also still make sense, which now it clearly doesn't anymore (see https://developer.wordpress.org/reference/functions/remove_theme_support/, which basically encourages to use remove_theme_support on those self-negating features.

So, in other words, you have to do linguistical and logical mind juggling to understand what what does.

Can't something be done about this?

🤷

**Comment by ciampo** [comment]
Hey @smileBeda ,

I'm not sure that the point you're raising is relevant to this specific issue. I would encourage you to open a separate issue or discussion, while also adopting a spirit of collaboration and assuming positive intent from the open source community.
